### ISSUE TITLE
foomatic-rip postscript preprocessing broken by #200

### ISSUE THREAD

**Issue by LynxChaus** [OpenPrinting/cups-filters#206]
After fixing #200 - postscript preprocessing is broken. Removing return at filter/foomatic-rip/postscript.c:1081 fix it.

**Comment by OdyX** [comment]
I can confirm this; foo2zjs doesn't print anything anymore since 1.27.1; that's a serious regression.

I have therefore included @LynxChaus 's patch and will send a pull-request when tested.

I'll change Debian's CUPS autopkgtests to detect these regressions too.

**Comment by tillkamppeter** [comment]
What was the exact problem here? Did all PostScript files sent through foomatic-rip come out empty (zero bytes) instead of the document in the printer's format? Or did only zero-page PostScript files come out empty and the printer did not like an empty job in some form.

**Comment by tillkamppeter** [comment]
And do all attempts to pass a PostScript file through foomatic-rip end up with a message "No pages left, outputting empty file." in the error_log?

**Comment by OdyX** [comment]
The easiest way for me to reproduce this was to run this (with printer-driver-foo2zjs installed) with the latest test-drivers script from CUPS: https://salsa.debian.org/printing-team/cups/blob/debian/master/debian/tests/utils/test-drivers :
```
sudo ./debian/tests/utils/test-drivers -l "foo2zjs:0/ppd/foo2zjs/HP-LaserJet_Pro_P1102w.ppd"
```
Without this patch, I got:
```
* Driver foo2zjs:0/ppd/foo2zjs/HP-LaserJet_Pro_P1102w.ppd
 - Create test printer: done.
 - Print /usr/share/cups/data/topsecret.pdf: . Filter succeeded but resulting file is empty; aborting!
 - Interrupted (or finished), delete test printer: done.
```
. With the patch, I got:
```
* Driver foo2zjs:0/ppd/foo2zjs/HP-LaserJet_Pro_P1102w.ppd
 - Create test printer: done.
 - Print /usr/share/cups/data/topsecret.pdf: . done.
 - Print /usr/share/cups/data/form_russian.pdf: .. done.
 - Print /usr/share/cups/data/default.pdf: . done.
 - Print /usr/share/cups/data/secret.pdf: . done.
 - Print /usr/share/cups/data/standard.pdf: . done.
 - Print /usr/share/cups/data/unclassified.pdf: . done.
 - Print /usr/share/cups/data/classified.pdf: . done.
 - Print /usr/share/cups/data/form_english.pdf: .. done.
 - Print /usr/share/cups/data/confidential.pdf: . done.
 - Print /usr/share/cups/data/default-testpage.pdf: .. done.
 - Delete test printer: done.
 - Interrupted (or finished), delete test printer: done
```
So what happened is that the data sent to the printer is 0bytes long. I haven't checked the error_log.

**Comment by tillkamppeter** [comment]
An error_log in debug mode would be great, to help fixing this correctly, so that both regular printing and zero-page-job handling works.

**Comment by LynxChaus** [comment]
removing return is fast & stupid fix. Better properly set non-empty flag in right place.

**Comment by tillkamppeter** [comment]
@JaiLuthra1, it seems that your zero-page job support in foomatic-rip considers jobs with pages as zero-page. Could you have a look? Thanks.

**Comment by JaiLuthra1** [comment]
Sure, I will have a look.

**Comment by JaiLuthra1** [comment]
@tillkamppeter All but the last page were sent to the printer resulting in empty file for the single page tests. Fixed in #209

**Comment by tillkamppeter** [comment]
Jai, thank you very much. I have merged your Pull Request now.
